### PR TITLE
Readability and style fixes to user-data-cleanup

### PR DESCRIPTION
**Pull request by eobrain**
1.  Added eslint configuration copied from Angular.
2.  Fixed most eslint errors and warnings.
3.  Fixed Flychecker warnings of violations of Google style.
4.  Made demo app work with older browser, by removing `const` and arrow functions. 
5.  Did some minor code simplifications, and readability improvements, such as replacing
     some if-else-ifs with switch-cases and reformatting line-breaking in test code.

### CODE REVIEW

**Comment by eobrain** [comment]
Fixed conflicts with base branch.

**Comment by reachpov1** [comment]
Help

**Comment by nicolasgarnier** [comment]
For the ESlint config, you can extend the published Google rules. For instance, this is the **.eslintrc** file I usually use for web apps:

```json
{
  "extends": "google",
  "env": {
    "browser": true
  },
  "rules": {
    "camelcase": 0,
    "default-case": 0,
    "max-len": 0,
    "max-nested-callbacks": 0,
    "no-inline-comments": 0,
    "no-warning-comments": 0,
    "require-jsdoc": 0,
    "valid-jsdoc": 0,
    "no-implicit-coercion": [2, {
      "boolean": false,
      "number": true,
      "string": true
    }],
    "no-unused-vars": [2, {
      "vars": "all",
      "args": "after-used",
      "argsIgnorePattern": "^var_args$"
    }],
    "no-negated-condition": 0
  },
  "globals": {
    "$": true,
    "firebaseUi": true,
    "uiConfig": true,
    "jQuery": true,
    "ga": true,
    "firebase": true,
    "gapi": true
  }
}
```

Otherwise LGTM :)

**Comment by eobrain** [comment]
Changed eslint to use published Google rules and tweaked code to fix style errors found.

Also fixed bug found in initialization, and cleaned up HTML template of demo app a bit.

File: user-data-cleanup/public/main.js [hunk 1]
```diff
@@ -87,29 +86,35 @@ Demo.prototype.addDataDemo = function(body) {
...
+  };
   var body = {ref: '/accounts/' + this.user.uid + '/', content: cont};
   this.addDataDemo(body);
 };
 
 Demo.prototype.record = function(year) {
   this.addDataDemo({
       ref: '/record/' + this.user.uid,
-      content: {content: 'Record data', createYear: year}
+    content: {content: 'Record data', createYear: year},
```
Comment (nicolasgarnier) [inline user-data-cleanup/public/main.js]: little indent issue here :)

Comment (nicolasgarnier) [reply user-data-cleanup/public/main.js]: I'll fix it :)

**Comment by nicolasgarnier** [comment]
Looks good! :)

**Pull request merged**